src/commands/update.rs

```diff
@@ -0,0 +1,90 @@
+use clap::ArgMatches;
+use log::{debug, error, info};
+
+use std::process::exit;
+
+use crate::server;
+
+const EXIT_NO_UPDATE_AVAILABLE: i32 = 0;
+const EXIT_UPDATE_AVAILABLE: i32 = 1;
+
+enum Action {
+  Check,
+  Force,
+  Regular,
+}
+
+impl Action {
+  fn new(check: bool, force: bool) -> Self {
+    match (check, force) {
+      (true, true) => panic!("`check` and `force` are mutually exlusive!"),
+      (true, false) => Self::Check,
+      (false, true) => Self::Force,
+      (false, false) => Self::Regular,
+    }
+  }
+}
+
+pub fn invoke(args: &ArgMatches) {
+  info!("Checking for updates");
+
+  if !server::is_installed() {
+    error!(
+      "Failed to find server executable. Can't update! If the server isn't installed yet then you \
+        likely need to run `odin install`."
+    );
+    exit(1);
+  }
+
+  let dry_run = args.is_present("dry_run");
+  let check = args.is_present("check");
+  let force = args.is_present("force");
+
+  let server_running = server::is_running();
+  let update_available = server::update_is_available();
+  if update_available {
+    info!("A server update is available!");
+  } else {
+    info!("No server updates found");
+  }
+
+  match Action::new(check, force) {
+    Action::Check => match (dry_run, update_available) {
+      (true, true) => {
+        info!("Dry run: An update is available. This would exit with 0 to indicate this.")
+      }
+      (true, false) => {
+        info!("Dry run: No update is available. This would exit with 1 to indicate this.")
+      }
+      (false, true) => exit(EXIT_NO_UPDATE_AVAILABLE),
+      // TODO: should we do a value other than 1 here, and if we do then what value?
+      (false, false) => exit(EXIT_UPDATE_AVAILABLE),
```

Comment:
Exit 1 if an update is available is fine, unless we want to use exit code 10 and document that it means there is an update available? 

Ideally, can we include an `info` log for if an update is available or not available?

Reply:
> Exit 1 if an update is available is

The only reason I mention possibly using an exit code is so that people can differentiate between no update available from other errors that may make `odin` exit 1. I'm not really sure how common that would be

> Ideally, can we include an info log for if an update is available or not available?

And there's a log above the match on `Action` :D

```rust
if update_available {
  info!("A server update is available!");
} else {
  info!("No server updates found");
}
```